### PR TITLE
chore: update Codecov settings

### PR DESCRIPTION
**Pull request by iuioiua**
Codecov isn't correctly updating commits (see [here](https://app.codecov.io/gh/denoland/deno_std/commits)). I've also noticed that the files changed are way off (see [here](https://app.codecov.io/gh/denoland/deno_std/pull/4620)). So Codecov may've been broken for 3 months. This change might fix that.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,9 +60,10 @@ jobs:
 
       - name: Upload coverage
         uses: codecov/codecov-action@v4
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (iuioiua) [inline .github/workflows/ci.yml:64]: @kt3k, please ensure this token has been added ([documentation](https://docs.codecov.com/docs/adding-the-codecov-token#github-actions)).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -60,9 +60,10 @@ jobs:
...
         with:
           name: ${{ matrix.os }}-${{ matrix.deno }}
-          files: cov.lcov
```
Comment (iuioiua) [inline .github/workflows/ci.yml:65]: Codecov targets this file by default anyway.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kt3k) [reply .github/workflows/ci.yml:64]: This is already set

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**